src/framework/uploadImage/PreviewImage.jsx

```diff
@@ -0,0 +1,33 @@
+
+import PropTypes from 'prop-types';
+import React from 'react';
+
+class PreviewImage extends React.Component {
+  componentDidUpdate(prevProps) {
+    if (this.props.imageBinaryUrl === prevProps.imageBinaryUrl) {
+      return null;
+    }
+    const image = new Image();
+
+    image.onload = () => {
+      this.props.hasLoaded(image);
+    };
+
+    image.src = this.props.imageBinaryUrl;
+  }
+
+  render() {
+    const image = this.props.imageBinaryUrl ?
+      this.props.imageBinaryUrl :
+      this.props.children;
```

Comment:
I'm not sure if you need to distinguish, the `UploadImage` could just control as it knows what is happening.